src/mixins/itext_click_behavior.mixin.ts

```diff
@@ -1,14 +1,20 @@
-//@ts-nocheck
-import { ObjectEvents } from '../EventTypeDefs';
+import { RIGHT_CLICK } from '../constants';
+import { TPointerEvent, TPointerEventInfo } from '../EventTypeDefs';
 import { IPoint, Point } from '../point.class';
 import type { DragMethods } from '../shapes/Object/InteractiveObject';
-import { TPointerEvent, TransformEvent } from '../typedefs';
 import { stopEvent } from '../util/dom_event';
 import { invertTransform, transformPoint } from '../util/misc/matrix';
 import { DraggableTextDelegate } from './DraggableTextDelegate';
+import { ITextEvents } from './itext_behavior.mixin';
 import { ITextKeyBehaviorMixin } from './itext_key_behavior.mixin';
 
-export abstract class ITextClickBehaviorMixin<EventSpec extends ObjectEvents>
+function isRightClick(button?: number) {
+  return button === RIGHT_CLICK;
+}
```

Comment:
in canvas file we define this function
```
function checkClick(e: TPointerEvent, value: number) {
  return !!(e as MouseEvent).button && (e as MouseEvent).button === value - 1;
}
```
So on top of reusing it, i wonder why there we use `value - 1` and here were are using value directly.

Reply:
Answer i found later: because here we are not inspecting e.button directly.

Reply:
it was previously e.button.
`checkClick` is used on the native event before the synthetic event is fired by fabric. The synthetic event has the `button` attr set and is `e.button+1`. It is far from ideal that that values aren't identical. But that is what it is.
I thought we should use the button attr of the synthetic event or use `checkClick` and made it clear we are checking 
 to avoid a right click.

Reply:
Ok somethin weird is going on here.
The old code was introduced 6 years ago to guard against NON LEFT CLICKS.
That was the intention.
The code seems wrong in that regard since the value of button for left clicks is `0` and so checking for `!==1` was plain wrong.
On top of that UTs have mocked events so not really reliable.
To avoid mixing types with changes, i just kept the exact code that was there before, but what this code is doing should be investigated.

Reply:
here the context:
https://github.com/fabricjs/fabric.js/pull/3888/files#diff-ca85fc40679e7fc083b650fbb7e7851f09d84a83da4401ad2ec326f57e31febaR86